pkg/scheduler/apis/config/testing/compatibility_test.go

```diff
@@ -1794,3 +1795,284 @@ func TestPluginsConfigurationCompatibility(t *testing.T) {
+
+			defProf := sched.Profiles["default-scheduler"]
+			gotPlugins := defProf.ListPlugins()
+			if diff := cmp.Diff(tc.wantPlugins, gotPlugins); diff != "" {
+				t.Errorf("unexpected plugins diff (-want, +got): %s", diff)
+			}
+		})
+	}
+}
+
+func TestPluginsArgumentsCompatibility(t *testing.T) {
+	testcases := []struct {
+		name         string
+		pluginConfig []config.PluginConfig
+	}{
+		{
+			name:         "Default",
+			pluginConfig: nil,
+		},
+		{
+			name: "User customed",
+			pluginConfig: []config.PluginConfig{
+				{
+					Name: "NodeResourcesFit",
+					Args: runtime.Unknown{
+						Raw: []byte(`{
+							"ignoredResources": [
+								"foo",
+								"bar"
+							]
+						}`),
+					},
+				},
+				{
+					Name: "PodTopologySpread",
+					Args: runtime.Unknown{
+						Raw: []byte(`{
+							"defaultConstraints": [
+								{
+									"maxSkew": 1,
+									"topologyKey": "foo",
+									"whenUnsatisfiable": "DoNotSchedule"
+								},
+								{
+									"maxSkew": 10,
+									"topologyKey": "bar",
+									"whenUnsatisfiable": "ScheduleAnyway"
+								}
+							]
+						}`),
+					},
+				},
+				{
+					Name: "RequestedToCapacityRatio",
+					Args: runtime.Unknown{
+						Raw: []byte(`{
+							"shape":[
+								"Utilization": 5,
+								"Score": 5
+							],
+							"resources":[
+								"Name": "cpu",
+								"Weight": 10
+							]
+						}`),
+					},
+				},
+				{
+					Name: "InterPodAffinity",
+					Args: runtime.Unknown{
+						Raw: []byte(`{
+							"HardPodAffinityWeight": 100
+						}`),
+					},
+				},
+				{
+					Name: "NodeLabel",
+					Args: runtime.Unknown{
+						Raw: []byte(`{
+							"presentLabels": [
+								"foo",
+								"bar"
+							],
+							"absentLabels": [
+								"apple"
+							],
+							"presentLabelsPreference": [
+								"dog"
+							],
+							"absentLabelsPreference": [
+								"cat"
+							]
+						}`),
+					},
+				},
+				{
+					Name: "ServiceAffinity",
+					Args: runtime.Unknown{
+						Raw: []byte(`{
+							affinityLabels: [
+								"foo",
+								"bar"
+							],
+							antiAffinityLabelsPreference: [
+								"disk",
+								"flash"
+							]
+						}`),
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			client := fake.NewSimpleClientset()
+			informerFactory := informers.NewSharedInformerFactory(client, 0)
+			recorderFactory := profile.NewRecorderFactory(events.NewBroadcaster(&events.EventSinkImpl{Interface: client.EventsV1beta1().Events("")}))
+			_, err := scheduler.New(
+				client,
+				informerFactory,
+				informerFactory.Core().V1().Pods(),
+				recorderFactory,
+				make(chan struct{}),
+				scheduler.WithProfiles(config.KubeSchedulerProfile{
+					SchedulerName: v1.DefaultSchedulerName,
+					PluginConfig:  tc.pluginConfig,
+				}),
+				scheduler.WithBuildFrameworkCapturer(func(p config.KubeSchedulerProfile) {
+					if diff := cmp.Diff(tc.pluginConfig, p.PluginConfig); diff != "" {
```

Comment:
Remember that args could be different for different profiles.

Reply:
Sure, I will add other cases to cover this.

Reply:
not necessarily. I think you can cover most of it by merging these checks into `TestPluginsConfigurationCompatibility`